hatch/app.py

```diff
@@ -1,39 +1,54 @@
 import aiofiles
-from fastapi import Depends, FastAPI, Header, HTTPException, Security
+from fastapi import Depends, FastAPI, HTTPException, Security
 from fastapi.responses import FileResponse
 from fastapi.security.api_key import APIKeyHeader
-from pydantic import Required
+from pydantic import ValidationError
+from starlette.requests import Request
 
-from hatch import config, models
+from hatch import config, models, schema
+from hatch.signing import AuthToken, set_default_key
 
 
+app = FastAPI()
+# set key to use in signing
+set_default_key(config.BACKEND_TOKEN, config.BACKEND)
 api_key_header = APIKeyHeader(name="Authorization")
 
 
-def validate(auth_token: str = Security(api_key_header)):
-    if auth_token != config.BACKEND_TOKEN:
+def validate(request: Request, auth_token: str = Security(api_key_header)):
+    try:
+        token = AuthToken.verify(auth_token)
+    except ValidationError:
         raise HTTPException(403, "Unauthorised")
 
+    if not str(request.url).startswith(token.url):
```

Comment:
Can we put a comment above this check to say why they should match at the start?  Despite having this design explained to me multiple times (and agreeing with it!), I've forgotten this bit again!